src/backend/parser/gram.y

```diff
@@ -16959,6 +16962,23 @@ check_generic_type_with_or_without_time_zone(const TypeName *typname, bool with_
 	return false; /* not allowed */
 }
 
+/*
+ * Some T-SQL domain types, e.g., sys.sysname from PG13, can have wrong typmods
+ * so that it can cause a failure during dump and restore. This function detects
+ * problematic cases and fixes it.
+ */
+static void
+fix_tsql_domain_typmods(TypeName *typname)
+{
+	const char *dump_restore = GetConfigOption("babelfishpg_tsql.dump_restore", true, false);
+	if (!dump_restore || strcmp(dump_restore, "on") != 0)
+		return;
+
+	if (strcmp(strVal(linitial(typname->names)), "sys") == 0 &&
```

Comment:
have we verified that sys.sysname is only domain with this problem?

Reply:
The problem only happens when a domain’s base type has a fixed typmod, e.g., `varchar(128)`, and a function is created under psql, e.g., `sys.suser_sid(sys.sysname)`, and a customer-created view refers that function. So, as Rishabh mentioned, the only exception would be `sys._ci_sysname` though it is very unlikely...